### PR TITLE
boards/cc1312-launchpad: add OpenOCD configuration

### PR DESCRIPTION
**Pull request by jeandudey**
### Contribution description

This enables OpenOCD for the CC1312 LaunchPad.

All we need is:

- The Texas Instruments version of OpenOCD
  (https://git.ti.com/cgit/sdo-emu/openocd)
- Set the `PROGRAMMER` environment variable to `openocd`.
- Connect a `cc1312-launchpad` and play with the commands.

### CODE REVIEW

File: boards/cc1312-launchpad/dist/openocd.cfg [hunk 1]
```diff
@@ -0,0 +1,11 @@
+source [find interface/xds110.cfg]
+
+transport select jtag
+gdb_memory_map enable
+gdb_flash_program enable
+
+source [find target/ti_cc26x2.cfg]
```
Comment (benpicco) [inline boards/cc1312-launchpad/dist/openocd.cfg]: Looks like this is generic enough to move it to move it to `./common/cc26x2_cc13x2/dist/openocd.cfg`.

The you can probably add `OPENOCD_CONFIG ?= $(RIOTBOARD)/common/cc26x2_cc13x2/dist/openocd.cfg` to `boards/cc1352-launchpad/Makefile.include` too.

Comment (jeandudey) [reply boards/cc1312-launchpad/dist/openocd.cfg]: Thanks 😉!

File: boards/cc1352-launchpad/Makefile.include [hunk 2]
```diff
@@ -8,4 +8,11 @@ PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbmodem*)))
 include $(RIOTMAKE)/tools/serial.inc.mk
 
 # configure the flash tool
-include $(RIOTMAKE)/tools/uniflash.inc.mk
+PROGRAMMER ?= openocd
+
+ifeq ($(PROGRAMMER),uniflash)
+  include $(RIOTMAKE)/tools/uniflash.inc.mk
+else ifeq ($(PROGRAMMER),openocd)
+  OPENOCD_CONFIG ?= $(RIOTBOARD)/common/cc26x2_cc13x2/dist/openocd.cfg
```
Comment (benpicco) [inline boards/cc1352-launchpad/Makefile.include]: You can set `OPENOCD_CONFIG` unconditionally.

File: boards/cc1312-launchpad/Makefile.include [hunk 3]
```diff
@@ -8,4 +8,11 @@ PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbmodem*)))
...
 # configure the flash tool
-include $(RIOTMAKE)/tools/uniflash.inc.mk
+PROGRAMMER ?= uniflash
+
+ifeq ($(PROGRAMMER),uniflash)
+  include $(RIOTMAKE)/tools/uniflash.inc.mk
+else ifeq ($(PROGRAMMER),openocd)
+  OPENOCD_CONFIG ?= $(RIOTBOARD)/common/cc26x2_cc13x2/dist/openocd.cfg
+  include $(RIOTMAKE)/tools/openocd.inc.mk
+endif
```
Comment (benpicco) [inline boards/cc1312-launchpad/Makefile.include]: You can also move this to `boards/common/cc26x2_cc13x2/Makefile.include` and then just `include $(RIOTBOARD)/common/cc26x2_cc13x2/Makefile.include`

**Pull request merged**